plugins/woocommerce/src/Internal/Features/FeaturesController.php

```diff
@@ -668,53 +682,128 @@ public function change_feature_enable( string $feature_id, bool $enable ): bool
 	 * FeaturesUtil::declare_compatibility instead, passing the full plugin file path instead of the plugin name.
 	 *
 	 * @param string $feature_id Unique feature id.
-	 * @param string $plugin_name Plugin name, in the form 'directory/file.php'.
+	 * @param string $plugin_file Plugin file path, either full or in the form 'directory/file.php'.
 	 * @param bool   $positive_compatibility True if the plugin declares being compatible with the feature, false if it declares being incompatible.
 	 * @return bool True on success, false on error (feature doesn't exist or not inside the required hook).
 	 * @throws \Exception A plugin attempted to declare itself as compatible and incompatible with a given feature at the same time.
 	 */
-	public function declare_compatibility( string $feature_id, string $plugin_name, bool $positive_compatibility = true ): bool {
+	public function declare_compatibility( string $feature_id, string $plugin_file, bool $positive_compatibility = true ): bool {
 		if ( ! $this->proxy->call_function( 'doing_action', 'before_woocommerce_init' ) ) {
 			$class_and_method = ( new \ReflectionClass( $this ) )->getShortName() . '::' . __FUNCTION__;
 			/* translators: 1: class::method 2: before_woocommerce_init */
 			$this->proxy->call_function( 'wc_doing_it_wrong', $class_and_method, sprintf( __( '%1$s should be called inside the %2$s action.', 'woocommerce' ), $class_and_method, 'before_woocommerce_init' ), '7.0' );
 			return false;
 		}
-
 		if ( ! $this->feature_exists( $feature_id ) ) {
 			return false;
 		}
 
-		$plugin_name = str_replace( '\\', '/', $plugin_name );
+		if ( $this->lazy ) {
+			// Lazy mode: Queue to be normalized later.
+			$this->pending_declarations[] = array( $feature_id, $plugin_file, $positive_compatibility );
+			return true;
+		}
+
+		// Late call: Normalize and register immediately.
+		$plugin_id = $this->plugin_util->get_wp_plugin_id( $plugin_file );
+		if ( ! $plugin_id ) {
+			$this->proxy->call_function( 'wc_get_logger' )->error( "Invalid plugin file: {$plugin_file}" );
+			return false;
+		}
```

Comment:
Would it make sense to move this to `register_compatibility_internal()` given it's present in `process_pending_declarations()`?